developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,300 @@
+assert(bytes == [5, 0x61, 0x70, 0x74, 0x6F, 0x73]);
+```
+
+### Signing message
+
+A signing message is the bytes of the BCS serialized raw transaction. There is also a prefix, `prefix_bytes`, which is `sha3_256("APTOS::RawTransaction")`. Therefore:
+
+`signing_message = prefix_bytes + bcs_bytes_of_raw_transaction.`
+
+The below code shows how prefix bytes are calculated.
+
+```typescript
+function signinMessagePrefix(): Buffer {
+  let hash = SHA3.sha3_256.create();
+  hash.update(`APTOS::RawTransaction`);
+  return Buffer.from(hash.arrayBuffer());
+}
+```
+
+### Signature
+A signature is the [Ed25519][ed25519] encryption of a signing message with the client private key. The signature is mainly used for security purpose.
+
+* By signing a signing message with the private key, clients prove to the Aptos Blockchain that they have authorized the transaction be executed.
+* Aptos Blockchain will validate the signature with client account's public key to ensure that the transaction submitted is indeed signed by the client.
+
+### Signed transaction
+
+A signed transaction consisted of:
+
+- A raw transaction, and
+- An authenticator. The authenticator contains a client's public key and the signature of the raw transaction.
+
+After BCS serialization and hex-coding, signed transactions are ready for submission to the [Aptos REST interface](https://fullnode.devnet.aptoslabs.com/spec.html).
+
+### Multisignature transactions
+
+The Aptos Blockchain supports several signing methods for transactions, including the single signature, the K-of-N multisig, and more.
+
+A K-of-N multisig transaction means that for such a transaction to be executed, at least K out of the N authorized signers have signed the transaction and passed the check conducted by the chain.
+
+Transaction signatures are wrapped in `Authenticator`. The Aptos Blockchain validates the transactions submitted by clients by using the Authenticator data.  See a few examples below:
+
+In Typescript, this is how a single signer authenticator looks like:
+```typescript
+interface Authenticator {
+  public_key: Uint8Array,
+  signature: Uint8Array
+}
+```
+
+A multisig authenticator example is shown below:
+```typescript
+interface  MultiEd25519PublicKey {
+  public_keys: Uint8Array[], // A list of public keys
+  threshold: Uint8, // At least `threshold` signatures must be valid
+}
+
+interface MultiEd25519Signature {
+    signatures: Uint8Array[], // A list of signatures
+    // 4 bytes, at most 32 signatures are supported.
+    // If Nth bit value is `1`, the Nth signature should be provided in `signatures`. Bits are read from left to right
+    bitmap: Uint8Array,
+}
+
+interface MultisigAuthenticator {
+  public_key: MultiEd25519PublicKey,
+  signature: MultiEd25519Signature
+}
+```
+
+## Detailed steps
+
+1. Create a RawTransaction.
+2. Create a signing message.
+3. Sign the signing message with account private key.
+4. Create a SignedTransaction.
```

Comment:
feels like steps 2-4 could be compressed into
preparing the message to be signed, signing it, and generating an authenticator
or at least tweaking the phrasing

Reply:
Will consolidate 2 and 3 into one